viziphant/unit_tests/unit_test_unitary_event_analysis_plot.py

```diff
@@ -0,0 +1,329 @@
+            "unit_tests/target_images/target_image5_plot_coincidence"
+            "_rates.png")
+        cls.path_target_image5_plot_coincidence_rates = "target_images/" \
+            "target_image5_plot_coincidence_rates.png"
+
+        # create 6.figure with 1 subplot with plot_statistical_significance():
+        cls.target_image6_plot_statistical_significance = plt.figure(
+            "6", figsize=(20, 20))
+        plot_statistical_significance(
+            cls.spiketrains, cls.UE, ue.jointJ(0.05), window_size=100 * pq.ms,
+            window_step=10 * pq.ms, n_neurons=2,
+            plot_params_user={'fsize': 14},
+            plot_markers_user={'data_markercolor': ("k", "b", "r")},
+            position=(1, 1, 1))
+        cls.target_image6_plot_statistical_significance.savefig(
+            "/home/kramer/Documents/Studium Angewandte Mathematik und "
+            "Informatik/INM6/PycharmProjects/viziphant-new/viziphant/"
+            "unit_tests/target_images/target_image6_plot_statistical"
+            "_significance.png")
+        cls.path_target_image6_plot_statistical_significance = "target_images"\
+            "/target_image6_plot_statistical_significance.png"
+
+        # create 7.figure with 1 subplot with plot_unitary_events():
+        cls.target_image7_plot_unitary_events = plt.figure("7",
+                                                           figsize=(20, 20))
+        plot_unitary_events(
+            cls.spiketrains, cls.UE, ue.jointJ(0.05), binsize=5 * pq.ms,
+            window_size=100 * pq.ms, window_step=10 * pq.ms, n_neurons=2,
+            plot_params_user={'fsize': 14}, plot_markers_user={},
+            position=(1, 1, 1))
+        cls.target_image7_plot_unitary_events.savefig(
+            "/home/kramer/Documents/Studium Angewandte Mathematik und "
+            "Informatik/INM6/PycharmProjects/viziphant-new/viziphant/"
+            "unit_tests/target_images/target_image7_plot_unitary_events.png")
+        cls.path_target_image7_plot_unitary_events = "target_images/" \
+            "target_image7_plot_unitary_events.png"
+
+    def test_plot_unitary_event_full_analysis(self):
+        # create result image
+        result_image1_plot_UE_full_analysis = plt.figure("1.1",
+                                                         figsize=(20, 20))
+        plot_unitary_event_full_analysis(
+            self.spiketrains, self.UE, ue.jointJ(0.05), binsize=5 * pq.ms,
+            window_size=100 * pq.ms, window_step=10 * pq.ms, n_neurons=2,
+            plot_params_user={'fsize': 14},
+            plot_markers_user=[{},
+                               {'data_markercolor': ("r", "b")},
+                               {},
+                               {'data_markercolor': ("m", "g")},
+                               {'data_markercolor': ("k", "b", "r")},
+                               {}],
+            position=((6, 1, 1), (6, 1, 2), (6, 1, 3), (6, 1, 4), (6, 1, 5),
+                      (6, 1, 6)))
```

Comment:
as I said, this is something we don't want users to type.
this is an example of non-flexible API which becomes hard to maintain and develop in a year.